tests/test_ge_utils.py

```diff
@@ -164,8 +164,8 @@ def test_validate_invalid_parameters(
         ge.validate(dataset)
 
 
-def test_gen_directory_tree_str(tmp_path_factory):
-    project_dir = str(tmp_path_factory.mktemp("project_dir"))
+def test_gen_directory_tree_str(tmpdir):
+    project_dir = str(tmpdir.mkdir("project_dir"))
```

Comment:
`tmp_path_factory` is session scoped while `tmpdir` is function scoped.

Since this is a unit test, I don't believe we should be sharing state between various funcs. This is the reason why this test kept failing in the pipeline.